zebedee-cms/src/main/java/com/github/onsdigital/zebedee/service/DatasetService.java

```diff
@@ -4,29 +4,33 @@
 import com.github.onsdigital.zebedee.dataset.api.exception.DatasetNotFoundException;
 import com.github.onsdigital.zebedee.dataset.api.exception.UnexpectedResponseException;
 import com.github.onsdigital.zebedee.exceptions.ZebedeeException;
-import com.github.onsdigital.zebedee.json.CollectionInstance;
+import com.github.onsdigital.zebedee.json.CollectionDataset;
+import com.github.onsdigital.zebedee.json.CollectionDatasetVersion;
 
-import javax.management.InstanceNotFoundException;
 import java.io.IOException;
 
+/**
+ * Provides high level dataset functionality
+ */
 public interface DatasetService {
 
     /**
-     * Add an instance for the given instanceID to the collection for the collectionID.
-     * @param collectionID
-     * @param instanceID
-     * @return
-     * @throws ZebedeeException
-     * @throws IOException
+     * Update a dataset for the given ID to the collection for the given collectionID.
      */
-    CollectionInstance addInstanceToCollection(String collectionID, String instanceID) throws ZebedeeException, IOException, UnexpectedResponseException, InstanceNotFoundException, BadRequestException, DatasetNotFoundException;
+    CollectionDataset updateDatasetInCollection(String collectionID, String datasetID, CollectionDataset updatedDataset) throws ZebedeeException, IOException, UnexpectedResponseException, DatasetNotFoundException, BadRequestException;
```

Comment:
Instead of throwing `UnexpectedResponseException, DatasetNotFoundException, BadRequestException` just use `DatasetAPIException`

Reply:
Done